src/Components/Diagnosis/DiagnosesListAccordion.tsx

```diff
@@ -0,0 +1,107 @@
+import {
+  ActiveConditionVerificationStatuses,
+  ConditionVerificationStatus,
+  ConsultationDiagnosis,
+} from "./types";
+import { useTranslation } from "react-i18next";
+import { classNames, compareBy } from "../../Utils/utils";
+import { useState } from "react";
+import CareIcon from "../../CAREUI/icons/CareIcon";
+import ButtonV2 from "../Common/components/ButtonV2";
+
+interface Props {
+  diagnoses: ConsultationDiagnosis[];
+}
+
+type GroupedDiagnoses = Record<
+  ConditionVerificationStatus,
+  ConsultationDiagnosis[]
+>;
+
+function groupDiagnoses(diagnoses: ConsultationDiagnosis[]) {
+  const groupedDiagnoses = {} as GroupedDiagnoses;
+
+  for (const status of ActiveConditionVerificationStatuses) {
+    groupedDiagnoses[status] = diagnoses
+      .filter((d) => d.verification_status === status)
+      .sort(compareBy("is_principal"));
+  }
+
+  return groupedDiagnoses;
+}
+
+export default function DiagnosesListAccordion(props: Props) {
+  const [isVisible, setIsVisible] = useState(true);
+  const diagnoses = groupDiagnoses(props.diagnoses);
+
+  return (
+    <div>
+      <div className="flex justify-between">
+        {!isVisible && (
+          <ButtonV2
+            className="text-md w-full p-0 font-semibold text-black"
+            ghost
+            onClick={() => {
+              setIsVisible((prev) => !prev);
+            }}
+          >
+            <CareIcon icon="l-angle-down" className="h-7" />
+            Expand Diagnoses
+          </ButtonV2>
+        )}
+      </div>
+      <div
+        className={classNames("transition-all duration-500 ease-in-out")}
+        style={
+          isVisible
+            ? {
+                overflow: "visible",
+              }
+            : { height: "0px", overflow: "hidden" }
+        }
```

Comment:
Was this not possible with tailwind?

Reply:
Definitely possible 😅. I Initially reused this code from the `AccordionV2` component. I've made it work with the tailwind classes now, along with the same change for the `AccordionV2` component too.